packages/contentstack-import/src/import/modules/extensions.ts

```diff
@@ -109,19 +125,88 @@ export default class ImportExtensions extends BaseClass {
     );
   }
 
-  /**
-   * @method serializeExtensions
-   * @param {ApiOptions} apiOptions ApiOptions
-   * @returns {ApiOptions} ApiOptions
-   */
-  serializeExtensions(apiOptions: ApiOptions): ApiOptions {
-    const { apiData: extension } = apiOptions;
-    if (this.extUidMapper.hasOwnProperty(extension.uid)) {
-      log(this.importConfig, `Extension '${extension.title}' already exists. Skipping it to avoid duplicates!`, 'info');
-      apiOptions.entity = undefined;
-    } else {
-      apiOptions.apiData = extension;
-    }
-    return apiOptions;
+  async replaceExtensions(): Promise<any> {
+    const onSuccess = ({ response, apiData: { uid, title } = { uid: null, title: '' } }: any) => {
+      this.extSuccess.push(response);
+      this.extUidMapper[uid] = response.uid;
+      log(this.importConfig, `Extension '${title}' replaced successfully`, 'success');
+      fsUtil.writeFile(this.extUidMapperPath, this.extUidMapper);
+    };
+
+    const onReject = ({ error, apiData }: any) => {
+      this.extFailed.push(apiData);
+      log(this.importConfig, `Extension '${apiData.title}' failed to replace ${formatError(error)}`, 'error');
+      log(this.importConfig, error, 'error');
+    };
+
+    await this.makeConcurrentCall(
+      {
+        apiContent: this.existingExtensions,
+        processName: 'Replace extensions',
+        apiParams: {
+          reject: onReject.bind(this),
+          resolve: onSuccess.bind(this),
+          entity: 'update-extensions',
+          includeParamOnCompletion: true,
+        },
+        concurrencyLimit: this.importConfig.concurrency || this.importConfig.fetchConcurrency || 1,
+      },
+      this.replaceExtensionHandler.bind(this),
+    );
+  }
+
+  async replaceExtensionHandler({
+    apiParams,
+    element: extension,
+    isLastRequest,
+  }: {
+    apiParams: ApiOptions;
+    element: Record<string, string>;
+    isLastRequest: boolean;
+  }) {
+    return new Promise(async (resolve, reject) => {
+      const { items: [extensionsInStack] = [] }: any = await this.stack
+        .extension()
+        .query({ query: { title: extension.title } })
+        .findOne()
+        .catch((error) => {
+          apiParams.reject({
+            error,
+            apiData: extension,
+          });
+          reject(true);
+        });
```

Comment:
[MINOR] Can we take this block of code out of Promise? so that we do need not to use async inside the promise which will throw a lint issue.

Reply:
@antonyagustine , its not giving any lint issues as such, will consider this point on phase 2 as it takes more time to develop and test